packages/smooth_app/lib/pages/product/common/product_query_page.dart

```diff
@@ -373,72 +374,64 @@ class _ProductQueryPageState extends State<ProductQueryPage>
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    final List<String> messages = <String>[];
+
+    String counting = appLocalizations.user_list_length(
+      _model.supplier.partialProductList.totalSize,
+    );
+
+    final String? translatedCountry = _getTranslatedCountry();
+
+    if (pagedProductQuery.hasDifferentCountryWorldData()) {
+      if (pagedProductQuery.world) {
+        counting += ' (${appLocalizations.world_results_label})';
+      } else {
+        if (translatedCountry != null) {
+          counting += ' ($translatedCountry)';
         }
-        return SizedBox(
-          width: double.infinity,
-          child: SmoothCard(
-            child: Padding(
-              padding: const EdgeInsets.all(SMALL_SPACE),
-              child: Row(
-                children: <Widget>[
-                  Expanded(child: Text(messages.join('\n'))),
-                  if (pagedProductQuery.getWorldQuery() != null)
-                    _getIconButton(
-                      _getWorldAction(
-                        appLocalizations,
-                        worldQuery!,
-                        widget.includeAppBar,
-                      ),
-                    ),
-                ],
-              ),
-            ),
+      }
+    }
+
+    messages.add(counting);
+
+    final int? lastUpdate = _model.supplier.timestamp;
+    if (lastUpdate != null) {
+      final String lastTime =
+          ProductQueryPageHelper.getDurationStringFromTimestamp(
+              lastUpdate, context);
+      messages.add('${appLocalizations.cached_results_from} $lastTime');
+    }
+
+    return SizedBox(
+      width: double.infinity,
+      child: SmoothCard(
+        child: Padding(
+          padding: const EdgeInsets.all(SMALL_SPACE),
+          child: Row(
+            children: <Widget>[
+              Expanded(child: Text(messages.join('\n'))),
+              if (pagedProductQuery.getWorldQuery() != null)
+                _getIconButton(
+                  _getWorldAction(
+                    appLocalizations,
+                    worldQuery!,
+                    widget.includeAppBar,
+                  ),
+                ),
+            ],
           ),
-        );
-      },
+        ),
+      ),
     );
   }
 
-  Future<String?> _getTranslatedCountry() async {
+  String? _getTranslatedCountry() {
     if (_country == null) {
       return null;
     }
-    final String locale = Localizations.localeOf(context).languageCode;
-    final List<Country> localizedCountries =
-        await IsoCountries.isoCountriesForLocale(locale);
-    for (final Country country in localizedCountries) {
-      if (country.countryCode.toLowerCase() == _country?.offTag.toLowerCase()) {
-        return country.name;
-      }
-    }
-    return null;
+    final OpenFoodFactsLanguage locale = ProductQuery.getLanguage();
+    return _country!.getLocalizedName(locale);
```

Comment:
```suggestion
    return _country!.getLocalizedName(ProductQuery.getLanguage());
```
Generally speaking, good naming of variable is critical.
E.g., `locale` doesn't mean the same as `language`: think about the reviewer or anybody maintaining the code.